### PR TITLE
fix: use valid version of yq

### LINKED ISSUE #1054: [testinfra] Choco 0.10.15 failing to install packages in certain Windows versions
Verification of the APM CI workers is failing due to the installation of `yq:3.3` in certain Windows boxes: 
- windows-2012-r2-immutable
- windows-2012-r2-latest-immutable
- windows-2016-immutable
- windows-2016-latest-immutable
- windows-2019-latest-immutable 

OTOH, it passes for:
- windows-2019-docker-immutable
- windows-2019-immutable
- windows-2019-test-immutable

## Error
Screenshot 2021-03-24 at 18 26 35

## Related links
- Example errored pipeline: https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-test-pipeline-mbp/detail/master/178/pipeline/414
- Relates https://github.com/chocolatey/choco/issues/2040

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It modifies the `yq` version, from `3.3` to `3.3.4`. As you can see in https://chocolatey.org/packages/yq#versionhistory, there is no "3.3" version, but "3.3.2" and "3.3.4".

What I'm not sure about, is why the tests pass in certain versions of Windows, and in a consistent manner. Could it be possible that the repositories for those Windows versions do have the `3.3` version? Weird.

## Why is it important?
The pipeline that validates the workers is currently broken, as described in #1054

## Related issues
Closes #1054

### CODE REVIEW

**Comment by mdelapenya** [comment]
Currently triggering a build for the test pipeline or these changes here: https://apm-ci.elastic.co/blue/organizations/jenkins/apm-shared%2Fapm-test-pipeline-mbp/detail/PR-1055/1/pipeline

**Comment by mdelapenya** [comment]
Unfortunately, this change is not fixing the issue with the Windows servers. Need to investigate it more deeply.

**Comment by mdelapenya** [comment]
I've seen the following message on CI, on those OS flavours that pass:

**Windows-2019-test-immutable**
>[2021-03-24T20:32:43.730Z] Getting latest yq version for 3.3.4 ...
[2021-03-24T20:32:47.015Z] 
[2021-03-24T20:32:47.015Z] yq 3.3.4 [Approved] Downloads cached for licensed users
[2021-03-24T20:32:51.200Z] Installing yq version: 3.3.4 ...
[2021-03-24T20:32:51.201Z] Chocolatey v0.10.11
[2021-03-24T20:32:51.201Z] Installing the following packages:
[2021-03-24T20:32:51.201Z] yq
[2021-03-24T20:32:51.201Z] By installing you accept licenses for the packages.
[2021-03-24T20:32:55.391Z] 
[2021-03-24T20:32:55.391Z] yq v3.3.4 [Approved]

**Windows-2019-latest-immutable**
>[2021-03-24T20:09:05.082Z] Getting latest yq version for 3.3.4 ...
[2021-03-24T20:09:11.661Z] Installing yq version:  ...
[2021-03-24T20:09:12.233Z] Chocolatey v0.10.15
[2021-03-24T20:09:12.233Z] Install Command
[2021-03-24T20:09:12.233Z] 
[2021-03-24T20:09:12.233Z] Installs a package or a list of packages (sometimes specified as a
[2021-03-24T20:09:12.233Z]  packages.config). Some may prefer to use `cinst` as a shortcut for
[2021-03-24T20:09:12.233Z]  `choco install`.
[2021-03-24T20:09:12.233Z] 
[2021-03-24T20:09:12.233Z] NOTE: 100% compatible with older chocolatey client (0.9.8.32 and below)
[2021-03-24T20:09:12.233Z]  with options and switches. Add `-y` for previous behavior with no
[2021-03-24T20:09:12.233Z]  prompt. In most cases you can still pass options and switches with one
[2021-03-24T20:09:12.233Z]  dash (`-`). For more details, see the command reference (`choco -?`).
[2021-03-24T20:09:12.233Z] 
[2021-03-24T20:09:12.233Z] Usage
[2021-03-24T20:09:12.233Z] 
[2021-03-24T20:09:12.233Z]     choco install <pkg|packages.config> [<pkg2> <pkgN>] [<options/switches>]
[2021-03-24T20:09:12.233Z]     cinst <pkg|packages.config> [<pkg2> <pkgN>] [<options/switches>]

In https://gitter.im/chocolatey/chocolatey.org?at=57697d8a0ede04dc4903616a, they mention the following:

>Downloads cached for licensed users means if you have a Pro license, the install file is cached on a private CDN so you can still install even if the download URL the install script uses to retrieve the file changes or their server is down.

Could it be possible that the Windows machines are not enrolled in this Pro acccount?

File: resources/JenkinsfileTemplate.groovy [hunk 1]
```diff
@@ -262,7 +262,7 @@ pipeline {
               warnError('installTools failed')
             }
             steps {
-              installTools([ [tool: 'yq', version: '3.3' ] ])
+              installTools([ [tool: 'yq', version: '3.3.4' ] ])
```
Comment (mdelapenya) [inline resources/JenkinsfileTemplate.groovy:265]: It's not the version, as the `3.3` is resolved as `3.3.4`

File: src/test/groovy/InstallToolsStepTests.groovy [hunk 2]
```diff
@@ -93,7 +93,7 @@ class InstallToolsStepTests extends ApmBasePipelineTest {
     script.installTool([ tool: 'foo', version: 'x.y.z', provider: 'choco', extraArgs: "--foo 'bar' 'foo'" ])
     printCallStack()
     assertTrue(assertMethodCallContainsPattern('powershell', 'Install foo:x.y.z'))
-    assertTrue(assertMethodCallContainsPattern('powershell', """script=choco install foo --no-progress -y --version 'x.y.z' "--foo 'bar' 'foo'" """))
+    assertTrue(assertMethodCallContainsPattern('powershell', """script=choco install foo --no-progress -y --version='x.y.z' "--foo 'bar' 'foo'" """))
```
Comment (mdelapenya) [inline src/test/groovy/InstallToolsStepTests.groovy:96]: This commit must be reverted too

**Comment by kuisathaverat** [comment]
You can test it by using the template pipeline, it  has some test to install tools, indeed it is falling to install yq 3.3 for a while

https://apm-ci.elastic.co/job/apm-shared/job/apm-test-pipeline-mbp/view/change-requests/
https://github.com/elastic/apm-pipeline-library/blob/master/resources/JenkinsfileTemplate.groovy

**Review by cachedout — APPROVED** [review]

**Comment by mdelapenya** [comment]
> You can test it by using the template pipeline, it has some test to install tools, indeed, it is falling to install yq 3.3 for a while

This is what I'm exactly doing https://github.com/elastic/apm-pipeline-library/pull/1055#issuecomment-806153229, triggering that pipeline for my PR. But it seems the `yq` package is consistently missing for the aforementioned windows versions.

File: resources/JenkinsfileTemplate.groovy (hunk 1 above)
Comment (v1v) [reply resources/JenkinsfileTemplate.groovy:265]: `installTools` uses the versions >= `3.3` so the patch versions are searched and gather the newest one that fulfils the version match.

Comment (v1v) [reply resources/JenkinsfileTemplate.groovy:265]: IIRC, `yq` is not required but only the validation of the `installTools` step:
- https://github.com/elastic/apm-pipeline-library/pull/804

`nodejs` was the initial implementation, but with the new versions then it did not work well, so I probably swapped to `yq` to use something more stable.

Maybe for simplicity, we can forget about the `installTools` step and remove it, or use another tool

Comment (mdelapenya) [reply resources/JenkinsfileTemplate.groovy:265]: I agree, we could use another tool that we already know is present in all package managers for platform

**Comment by mdelapenya** [comment]
I think this can be closed for now

**Pull request closed without merging**